File: _01_Languages/Kotlin/src/_03_clases/_06_Interfaces.kt
/*
 * 🧩 Enunciado:
 * Crea una interfaz `IVehiculo` con métodos `arrancar()` y `detener()`.
 * Implementa la interfaz en `Coche` y `Moto`.
 */

interface IVehiculo {
    fun arrancar()
    fun detener()
}

class Coche : IVehiculo {
    override fun arrancar() = println("🚗 El coche arranca.")
    override fun detener() = println("🚗 El coche se detiene.")
}

class Moto : IVehiculo {
    override fun arrancar() = println("🏍️ La moto arranca.")
    override fun detener() = println("🏍️ La moto se detiene.")
}

fun main() {
    val coche: IVehiculo = Coche()
    val moto: IVehiculo = Moto()

    coche.arrancar()
    moto.arrancar()
    coche.detener()
    moto.detener()
}

File: _01_Languages/CSharp/src/_02_poo/_06_Interfaces.cs
/*
 * 🧩 Enunciado:
 * Crea una interfaz `IVehiculo` con los métodos `Arrancar()` y `Detener()`.
 * Implementa la interfaz en dos clases: `Coche` y `Moto`.
 */

using System;

interface IVehiculo
{
    void Arrancar();
    void Detener();
}

class Coche : IVehiculo
{
    public void Arrancar() => Console.WriteLine("🚗 El coche arranca.");
    public void Detener() => Console.WriteLine("🚗 El coche se detiene.");
}

class Moto : IVehiculo
{
    public void Arrancar() => Console.WriteLine("🏍️ La moto arranca.");
    public void Detener() => Console.WriteLine("🏍️ La moto se detiene.");
}

class _06_Interfaces
{
    static void Main()
    {
        IVehiculo coche = new Coche();
        IVehiculo moto = new Moto();

        coche.Arrancar();
        moto.Arrancar();
        coche.Detener();
        moto.Detener();
    }
}
